clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -248,6 +249,22 @@ class GCEUtil {
     allMIGSInRegion
   }
 
+  static Autoscaler queryZonalAutoscaler(String projectName,
+                                         String zone,
+                                         String serverGroupName,
+                                         GoogleCredentials credentials) {
+    try {
+      return credentials.compute.autoscalers().get(projectName, zone, serverGroupName).execute()
+    } catch (GoogleJsonResponseException e) {
+      // 404 is thrown if the autoscaler does not exist in the given zone. Any other exception needs to be propagated.
+      if (e.getStatusCode() != 404) {
```

Comment:
For one reason or another, we verify that a `404` occurred due to the resource actually missing in [Terraform](https://github.com/hashicorp/terraform/blob/master/builtin/providers/google/resource_compute_instance.go#L560), might be helpful to do that here too.